src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpMetricSerializer.cs

```diff
@@ -466,14 +478,28 @@ private static int WriteExemplar(byte[] buffer, int writePosition, in Exemplar e
         int exemplarLengthPosition = writePosition;
         writePosition += ReserveSizeForLength;
 
-        // TODO: Need to serialize exemplar.FilteredTags and add unit tests.
+        foreach (var tag in exemplar.FilteredTags)
+        {
+            writePosition = WriteTag(buffer, writePosition, tag, ProtobufOtlpMetricFieldNumberConstants.Exemplar_Filtered_Attributes);
+        }
 
         writePosition = ProtobufSerializer.WriteDoubleWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Exemplar_Value_As_Double, value);
 
         var time = (ulong)exemplar.Timestamp.ToUnixTimeNanoseconds();
         writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.Exemplar_Time_Unix_Nano, time);
 
-        // TODO: Need to serialize exemplar.SpanID and exemplar.TraceId and add unit tests.
+        if (exemplar.SpanId != default)
+        {
+            writePosition = ProtobufSerializer.WriteTagAndLength(buffer, writePosition, SpanIdSize, ProtobufOtlpMetricFieldNumberConstants.Exemplar_Span_Id, ProtobufWireType.LEN);
```

Comment:
I recommend extracting this as a separate method as it is used in 2 places.

Reply:
I had this change locally but didn't commit it because i was concerned it would make the PR harder to review.
I just pushed it.